app/views/decidim/challenges/challenges/show.html.erb

```diff
@@ -37,30 +37,29 @@
         </div>
 
         <div class="definition-data__item sdg">
-          <span class="definition-data__title"><%= t("sdg", scope: "activemodel.attributes.challenge") %></span>
-          <%= t_sdg(@sdg) %><br/>
-          <div class="ods">
-              <div class="text-container show">
-                  <p><%= t(@sdg + ".logo.line1", scope: "decidim.components.sdgs") %></p>
-                  <p><%= t(@sdg + ".logo.line2", scope: "decidim.components.sdgs") %></p>
-              </div>
-              <img alt="Logo ODS <%= @sdg_index %>"
-                src="/assets/decidim/sdgs/ods-<%= @sdg_index %>.svg" class="challenge--view">
-          </div>
-        </div>
+          <% if @challenge.sdg_code %>
+            <span class="definition-data__title"><%= t("sdg", scope: "activemodel.attributes.challenge") %></span>
+            <%= t_sdg(@challenge.sdg_code) %><br/>
+            <div class="ods">
+                <div class="text-container show">
+                    <p><%= t(@challenge.sdg_code + ".logo.line1", scope: "decidim.components.sdgs") %></p>
+                    <p><%= t(@challenge.sdg_code + ".logo.line2", scope: "decidim.components.sdgs") %></p>
+                </div>
+                <img alt="Logo ODS <%= (1 + Decidim::Sdgs::Sdg.index_from_code(@challenge.sdg_code.to_sym)).to_s.rjust(2, "0") %>"
```

Comment:
Try to minimize business logic in the workforce please. 
You can solve this using some conditionals with the params that you may have prepared in the controller